### ISSUE TITLE
Arch Linux: Explicit python version required in binary

### ISSUE THREAD

**Issue by dsifford** [docker/compose#3082]
Hello,

Thanks for your hard work on this. 

Just as an FYI, in Arch distros `/usr/bin/env python` is python 3. This is causing some build issues (which, I assume is related to compose's partial python 3 support).

Is there any possibility of changing the shebang line in the [binary](https://github.com/docker/compose/blob/master/bin/docker-compose#L1) to explicitly call `#!/usr/bin/env python2`?

**Comment by dnephin** [comment]
I would rather fix any python 3 issues. At this time, there's only one known issue with python 3: #2342

Please provide more information about the errors you're seeing so we can fix them.

**Comment by dsifford** [comment]
@dnephin I like your style! 

The error message I was getting was this: `TypeError: a bytes-like object is required, not 'str'`.

I was able to generate this error in two ways:
1. `image: alpine` = error; `image: busybox` = no error
2. Using a github repo as a build context.

(Not able to post the entire stack trace at the moment, but I can chime back later with them if you need it)

**Comment by dnephin** [comment]
I think the `TypeError` is happening as part of error handling, and I guess we don't have an integration test which covers that error.  I suspect the error is `No command specified`, because there was no command set in the `docker-compose.yml` and the `alpine` base image doesn't set one (where as `busybox` sets `sh` as the default.

The relevant traceback is here:

```
Traceback (most recent call last):
  File "compose/.tox/py34/bin/docker-compose", line 9, in <module>
    load_entry_point('docker-compose==1.7.0.dev0', 'console_scripts', 'docker-compose')()
  File "compose/compose/cli/main.py", line 69, in main
    log_api_error(e)
  File "compose/compose/cli/main.py", line 92, in log_api_error
    if 'client is newer than server' in e.explanation:
TypeError: Type str doesn't support the buffer API
```

I'm going to be fixing this in #3015

For the github repo issue, is it the same as #3038 ? or do you have a stack trace for it?

**Comment by dsifford** [comment]
Ah, you're probably right Re: no command specified. Thanks for the clarification.

Regarding the git repo issue, I wasn't able to find an example in the docs on what format to use, so I just guessed and used the https url. In #3038 I see that he used the SSH url. Could that possibly be the reason? (The repo I was attempting to access was public)

**Comment by dnephin** [comment]
I just tried with `build: https://github.com/dnephin/docker-zookeeper.git` on python3.4 and that worked for me (although the build was cached). So it might be another error case.

**Comment by dnephin** [comment]
The type error should be fixed by #3156.  Let's open a new issue for any of the underlying problems.
